src/Components/Aspire.Npgsql.EntityFrameworkCore.PostgreSQL/AspireEFPostgreSqlExtensions.cs

```diff
@@ -66,15 +66,112 @@ public static partial class AspireEFPostgreSqlExtensions
+                if (settings.Retry)
+                {
+                    builder.EnableRetryOnFailure();
+                }
+                // 2. "Scale proportionally: You want to ensure that you don't scale out a resource to a point where it will exhaust other associated resources."
+                // The pooling is enabled by default, the min pool size is 0 by default: https://www.npgsql.org/doc/connection-string-parameters.html#pooling
+                // There is nothing for us to set here.
+                // 3. "Timeout: Places limit on the duration for which a caller can wait for a response."
+                // The timeouts have default values, except of Internal Command Timeout, which we should ignore:
+                // https://www.npgsql.org/doc/connection-string-parameters.html#timeouts-and-keepalive
+                // There is nothing for us to set here.
+            });
+            configureDbContextOptions?.Invoke(dbContextOptionsBuilder);
+        }
+    }
+
+    /// <summary>
+    /// Configures connection retries, health check, logging and telemetry for the <see cref="DbContext" />.
+    /// </summary>
+    /// <exception cref="ArgumentNullException">Thrown if mandatory <paramref name="builder"/> is null.</exception>
+    /// <exception cref="InvalidOperationException">Thrown when mandatory <see cref="DbContext"/> is not registered in DI.</exception>
+    public static void EnrichNpgsqlDbContext<[DynamicallyAccessedMembers(RequiredByEF)] TContext>(
+            this IHostApplicationBuilder builder,
+            Action<NpgsqlEntityFrameworkCorePostgreSQLSettings>? configureSettings = null) where TContext : DbContext
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+
+        NpgsqlEntityFrameworkCorePostgreSQLSettings settings = new();
+        var typeSpecificSectionName = $"{DefaultConfigSectionName}:{typeof(TContext).Name}";
+        var typeSpecificConfigurationSection = builder.Configuration.GetSection(typeSpecificSectionName);
+        if (typeSpecificConfigurationSection.Exists()) // https://github.com/dotnet/runtime/issues/91380
         {
-            builder.Services.AddDbContextPool<TContext>(ConfigureDbContext);
+            typeSpecificConfigurationSection.Bind(settings);
         }
         else
         {
-            builder.Services.AddDbContext<TContext>(ConfigureDbContext);
+            builder.Configuration.GetSection(DefaultConfigSectionName).Bind(settings);
         }
 
+        configureSettings?.Invoke(settings);
+
+        ConfigureRetry();
+
+        ConfigureInstrumentation<TContext>(builder, settings);
+
+        void ConfigureRetry()
+        {
+            if (!settings.Retry)
+            {
+                return;
+            }
+
+            // Resolving DbContext<TContextService> will resolve DbContextOptions<TContextImplementation>.
```

Comment:
I'm not sure I understand the significance of this line. What exactly does it mean?

Reply:
It's a reminder that we seek the service descriptor for `DbContextOptions<TestDbContext>` (referenced as TContextImplementation in EF APIs) because it is the db context options that is resolved even when we register a db context with an interface (TContextService in EF APIs)